JM/CF/Scripts/3_Game/CommunityFramework/Utils/CF_Localiser.c

```diff
@@ -0,0 +1,209 @@
+		Set(5, param6);
+		Set(6, param7);
+		Set(7, param8);
+		Set(8, param9);
+	}
+
+	static void Write(ParamsWriteContext ctx, CF_Localiser localiser)
+	{
+		ctx.Write(localiser.m_Count);
+
+		for (int i = 0; i < localiser.m_Count; i++)
+		{
+			ctx.Write(localiser.m_Strings[i]);
+			ctx.Write(localiser.m_Translates[i]);
+		}
+	}
+
+	static bool Read(ParamsReadContext ctx, inout CF_Localiser localiser)
+	{
+		if (!localiser) localiser = new CF_Localiser();
+
+		if (!ctx.Read(localiser.m_Count)) return false;
+
+		for (int i = 0; i < localiser.m_Count; i++)
+		{
+			string str;
+			bool translates;
+
+			if (!ctx.Read(str) || !ctx.Read(translates)) return false;
+
+			localiser.m_Strings[i] = str;
+			localiser.m_Translates[i] = translates;
+		}
+
+		return true;
+	}
+
+	void SetTranslates(bool translates)
+	{
+		m_Translates[0] = translates;
+	}
+
+	string Format()
+	{
+		return ToStr();
+	}
+
+	string ToStr()
+	{
+		string text = GetTranslated(-1);
+		string p1 = GetTranslated(0);
+		string p2 = GetTranslated(1);
+		string p3 = GetTranslated(2);
+		string p4 = GetTranslated(3);
+		string p5 = GetTranslated(4);
+		string p6 = GetTranslated(5);
+		string p7 = GetTranslated(6);
+		string p8 = GetTranslated(7);
+		string p9 = GetTranslated(8);
+
+		return string.Format(text, p1, p2, p3, p4, p5, p6, p7, p8, p9);
+	}
+
+	string GetText()
+	{
+		return Get(-1);
+	}
+
+	CF_Localiser Set(int index, bool value)
+	{
+		return Set(index, value.ToString(), false);
+	}
+
+	CF_Localiser Set(int index, int value)
+	{
+		return Set(index, value.ToString(), false);
+	}
+
+	CF_Localiser Set(int index, float value)
+	{
+		return Set(index, value.ToString(), false);
+	}
+
+	CF_Localiser Set(int index, vector value)
+	{
+		return Set(index, value.ToString(false), false);
+	}
+
+	CF_Localiser Set(int index, Class value)
+	{
+		string toStr = "";
+		g_Script.CallFunction(value, "ToStr", toStr, null);
+		return Set(index, toStr, false);
+	}
+
+	CF_Localiser Set(int index, string value)
+	{
+		return Set(index, value, true);
+	}
+
+	CF_Localiser Set(int index, string value, bool translates)
+	{
+		int realIndex = index + 1;
+		if (realIndex < 0 || realIndex > 9)
+			return this;
+
+		if (realIndex >= m_Count)
+		{
+			if (value == "" && realIndex != 0)
+				return this;
+
+			m_Count = realIndex + 1;
+		}
+
+		m_Strings[realIndex] = value;
+		m_Translates[realIndex] = translates;
+		return this;
+	}
+
+	string Get(int index)
+	{
+		int realIndex = index + 1;
+		if (realIndex < 0 || realIndex >= m_Count)
+			return "";
+		return m_Strings[realIndex];
+	}
+
+	string GetTranslated(int index)
+	{
+		int realIndex = index + 1;
+		if (realIndex < 0 || realIndex >= m_Count)
+			return "";
+		string value = m_Strings[realIndex];
+		if (m_Translates[realIndex])
+		{
+			value = Widget.TranslateString("#" + value);
+			if (value == "" || value[0] == " ")
+				return m_Strings[realIndex];
+		}
+		return value;
+	}
+
+	/* Backwards compatbility - to be removed after CF 1.4 */
```

Comment:
Please open an issue that references this code so we remember to address that.

Reply:
https://github.com/Jacob-Mango/DayZ-Community-Framework/issues/78